src/index.js

```diff
@@ -366,82 +450,70 @@ class RavenLambdaWrapper {
-							category: "lambda",
-							level: "info",
-							data: {}
-						};
-						if (event.requestContext) {
-							// Track HTTP request info as part of the breadcrumb
-							_.extend(breadcrumb.data, {
-								http_method: event.requestContext && event.requestContext.httpMethod,
-								host: event.headers && event.headers.Host,
-								path: event.path,
-								user_agent: event.headers && event.headers["User-Agent"]
-							});
-						}
-						Raven.captureBreadcrumb(breadcrumb);
+				if (pluginConfig.autoBreadcrumbs) {
+					// First breadcrumb is the invocation of the Lambda itself
+					const breadcrumb = {
+						message: process.env.AWS_LAMBDA_FUNCTION_NAME,
+						category: "lambda",
+						level: "info",
+						data: {}
+					};
+					if (event.requestContext) {
+						// Track HTTP request info as part of the breadcrumb
+						_.extend(breadcrumb.data, {
+							http_method:
+								event.requestContext && event.requestContext.httpMethod,
+							host: event.headers && event.headers.Host,
+							path: event.path,
+							user_agent: event.headers && event.headers["User-Agent"]
+						});
 					}
+					const Sentry = pluginConfig.sentryClient;
+					Sentry.addBreadcrumb(breadcrumb);
+				}
 
-					// And finally invoke the original handler code
-					const promise = handler(event, context, callback);
-					if (promise && _.isFunction(promise.then)) {
-						// don't forget to stop timers
-						return promise
+				// And finally invoke the original handler code
+				const promise = handler(event, context, callback);
+				if (promise && _.isFunction(promise.then)) {
+					// don't forget to stop timers
+					return promise
 						.then((...data) => {
 							clearTimers();
 							return Promise.resolve(...data);
 						})
 						.catch(err => {
 							clearTimers();
-							if (ravenInstalled && err && pluginConfig.captureErrors) {
-								const Raven = pluginConfig.ravenClient;
-								return new Promise((resolve, reject) => Raven.captureException(err, {}, () => {
-									reject(err);
-								}));
-							}
-							else {
-								return Promise.reject(err);
-							}
+							return Promise.reject(err);
 						});
-					}
-					// Returning non-Promise values would be meaningless for lambda.
-					// But inherit the behavior of the original handler.
-					return promise;
 				}
-				catch (err) {
-					// Catch and log synchronous exceptions thrown by the handler
-					captureUnhandled(err);
-				}
-			}, err => {
-				// Catch unhandled exceptions and rejections
-				if (!_.isObject(err) || err._ravenHandled) {
-					// This error is being rethrown. Pass it through...
-					throw err;
-				}
-				else {
-					captureUnhandled(err);
-				}
-			});
+				// Returning non-Promise values would be meaningless for lambda.
+				// But inherit the behavior of the original handler.
+				return promise;
+			}
+			catch (err) {
+				// Catch and log synchronous exceptions thrown by the handler
+				captureUnhandled(err);
```

Comment:
@aheissenberger  I'm testing it locally, and somehow my errors are not captured here.

Reply:
Shall captureUnhandled be in line 504 too?

Reply:
@ahanof 

> Shall captureUnhandled be in line 504 too?

no - the capture happens in the wrapper of the fail handler line 322 and 452.
Serverless offline or AWS Lambda will do a callback on "fail" when rejected.
Have a look at the tests to see what happens and to find out what's the problem here.

Do you use promises or callbacks in your lambda function?

Reply:
I'm using promises